macros/base/start_print.cfg

```diff
@@ -163,14 +163,47 @@ gcode:
     # Set vars
     {% set St = printer["gcode_macro _USER_VARIABLES"].travel_speed * 60 %}
     {% set Sz = printer["gcode_macro _USER_VARIABLES"].z_drop_speed * 60 %}
+    {% set verbose = printer["gcode_macro _USER_VARIABLES"].verbose %}
+
     {% set prime_line_x, prime_line_y = printer["gcode_macro _USER_VARIABLES"].prime_line_xy|map('float') %}
+    {% set prime_line_direction = printer["gcode_macro _USER_VARIABLES"].prime_line_direction|string|upper %}
     {% set prime_line_length = printer["gcode_macro _USER_VARIABLES"].prime_line_length|float %}
+    {% set prime_line_purge_distance = printer["gcode_macro _USER_VARIABLES"].prime_line_purge_distance|float %}
     {% set prime_line_flowrate = printer["gcode_macro _USER_VARIABLES"].prime_line_flowrate|float %}
     {% set klippain_ercf_enabled = printer["gcode_macro _USER_VARIABLES"].klippain_ercf_enabled %}
-    {% set verbose = printer["gcode_macro _USER_VARIABLES"].verbose %}
 
-    {% set max_extrude_cross_section = printer.configfile.settings.extruder.max_extrude_cross_section|float %}
-    {% set Sp = (prime_line_flowrate / max_extrude_cross_section) * 60 |float %}
+    {% set max_extrude_cross_section = printer["configfile"].config["extruder"]["max_extrude_cross_section"]|float %}
+    {% set filament_diameter = printer["configfile"].config["extruder"]["filament_diameter"]|lower %}
```

Comment:
should be a float instead of a string (lower)

Reply:
Oops, my bad, I'll fix that!

Reply:
C'est fait !